code-challenges.js

```diff
@@ -3,11 +3,28 @@
 
 // --------------------1) Create a function that takes in an array of objects and returns a sentence about each person with their names capitalized.
 
-var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy,"}, {name: "arthur dent", occupation: "radio employee"}]
+var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy"}, {name: "arthur dent", occupation: "radio employee"}]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is the president of the galaxy." "Arthus Dent is a radio employee."
 
-
-
+// create a function called prettyDescription
+prettyDescription = (arr) =>{
+    let capIt = arr.replace 
+    
+    let emptyArr = arr.map((value, index) => {
+        // return the name and captilize the first letters
+        // used regular expression to replace the first letter in key called name to go through each word  and replaced it with capIt which is a function to make it capitalized.
+        let cappedIt = value.name.replace(/\b\w/g, capIt => capIt.toUpperCase()) 
+        
+        // takes in an array of objects and combines the name and occupation 
+        if (index % 2 !== 0){
+            return `${cappedIt} is the ${value.occupation}`
+        } else{
+            return `${cappedIt} is a ${value.occupation}`
+        }
+    }) 
+    return emptyArr.toString()
+}
+console.log(prettyDescription(people));
```

Comment:
great job! so this solution works with _this_ array as far as "is the" vs "is a". how can you refactor this to work if we change the order of the array?